### DISCUSSION TITLE
How does Inertia handle Server Redirects

### DISCUSSION THREAD

**Discussion by bhaidar** [inertiajs/inertia#1230] [Help]
Hi,
I am trying to understand how Inertia handles server redirects.

In the docs, the following is mentioned:

```
Redirects

When making a non-GET Inertia request, via <Link> or manually, be sure to always respond with a proper Inertia response.

For example, if you're creating a new user, have your "store" endpoint return a redirect back to a standard GET endpoint, such as your user index page.

Inertia will automatically follow this redirect and update the page accordingly. Here's a simplified example.
```

For instance, in Laravel, when trying to reset the password, the user clicks on the "Email Reset Link". This action on the server does a `back()->with()`. I've studied this request inside DevTools and I can see a 302 Response with a Location response header.

It seems Inertia is intercepting this and requesting the Location but I am not able to understand where it's doing this in the source code.

Appreciate your help,

Bill

**Comment by bhaidar** [comment]
@reinink I appreciate your feedback :-)

**Comment by reinink** [comment]
Hey! The Inertia Laravel adapter automatically does a "redirect back" in the instance that no other response is returned from a controller endpoint. You can read more about this feature in our release notes, here: https://github.com/inertiajs/inertia-laravel/releases/tag/v0.6.0

**Reply by bhaidar** [reply]
Hey @reinink 
Thanks a lot for your feedback. I went through the PR, and I believe I understand the idea now on the backend when there is no response, InertiaJS considers this as a redirect back.

However, let's say I manually return the Redirect::back() (or even Inertia returns this). How's the Inertia on the JavaScript side handling this? 

In the example I gave above, I am POSTing a form, and the backend returns a Redirect::back() with a Response Status of 302. I then see a new request (generated by Inertia as it has the request header of x-inertia) is sent to the server with a URL the same as that of the URL returned by the server in the Location header.

I cannot find the code on the JavaScript that sends that request to the server as a result of Redirect::back(). 

Or is this process happening automatically by the browser? Could it be that the browser re-sends another request to the URL found in Location using the same state of the previous request, i.e., XMLHttpRequest, request header of x-inertia, etc.

Thanks :-)
